### ISSUE TITLE
[RFC] refactor based on l10n_it_ddt and l10n_it_edi from odoo SA

### ISSUE THREAD

**Issue by eLBati** [OCA/l10n-italy#1545]
Riprendo il discorso legato alla ristrutturazione del DDT, vedi https://github.com/OCA/l10n-italy/issues/1499, alla chiamata di stamattina e al fatto che Odoo SA sta lavorando ad un modulo `l10n_it_ddt`.

Vi riporto il feedback di @jco-odoo: https://github.com/odoo/odoo/pull/32156#issuecomment-559740467

Ora, la proposta da valutare è

> It would be good to have the modules of OCA being able to re-use the Odoo standard rather than define everything again and we can certainly use some help in making the Italian localization better

Sicuramente l'idea è interessante, visto che in generale avere dei moduli di localizzazione gestiti e mantenuti da Odoo SA (se ben fatti, col nostro aiuto) ridurrebbe i costi per il gruppo OCA Italia.

Considerato che il loro modulo `l10n_it_ddt` attualmente dipende da `l10n_it_edi` e comunque, se facciamo il progetto insieme, sarebbe bene farlo per tutti i moduli, seguire la loro proposta implicherebbe anche basarci sul loro `l10n_it_edi` per quanto riguarda la fatturazione elettronica.

Nota doverosa: se prima di fare il modulo `l10n_it_edi` ci avessero coinvolto, avremmo tutti risparmiato tempo. Questo glielo possiamo far notare.

Detto questo, alcune considerazioni.

Nel lavoro di ristrutturazione del modulo del DDT potremmo basarci sul loro `l10n_it_ddt` senza troppa fatica (dobbiamo comunque fare una ristrutturazione completa), soprattutto se accetteranno le nostre eventuali proposte di modifiche.

Ristrutturare tutti i moduli di fatturazione elettronica per basarci su `l10n_it_edi` sarebbe sicuramente più costoso.
Ma, visto che dovremo comunque affrontare una ristrutturazione dovuta a https://github.com/odoo/odoo/pull/33797, vale la pena farla basandoci su `l10n_it_edi`?
Farla basata su `l10n_it_edi` potrebbe comunque essere più costosa che farla sugli attuali moduli OCA, ma potrebbe portare più vantaggi a lungo termine.

Commenti?

**Comment by difra** [comment]
Assolutamente d'accordo. 

E la cosa interessante è che in questo modo viene lasciata aperta la strada della fatturazione da ordine, comunque compatibile con la gestione del DDT.

https://github.com/odoo-dev/odoo/blob/master-l10n-it-ddt-anr-jco/addons/l10n_it_ddt/models/account_invoice.py

**Comment by OpenCode** [comment]
Mi associo a quanto dice @difra. Però...
Quello che mi preoccupa è che la nostra scelta "strutturale" avrebbe un peso ancora minore di quello che ha adesso. Abbiamo visto nelle passate call che trovare la quadra tra di noi per comprendere quale sia il sistema migliore per impostare la struttura dei DDT è un filino complicato. Se in questa situazione entra in gioco anche un protagonista abbastanza sordo come Odoo SA allora la cosa si complica un po'.

Odoo ha l'abitudine di stravolge, noi siamo un po' più conservatori. La strada che stanno intraprendendo, noi l'abbiamo già usata con tutti i suoi limiti nel passato.

Servirebbe che si fermassero e ci coinvolgessero nella fase di analisi preliminare. Possiamo portare migliaia di ore complessive di esperienza sia tecnica che funzionale su un argomento di cui loro conoscono a mala pena il significato. IMHO.

**Comment by difra** [comment]
> Servirebbe che si fermassero e ci coinvolgessero nella fase di analisi preliminare. 

Vero, però prima meglio se ci chiudiamo in una stanza per una settimana a trovare una quadra finché non troviamo una linea comune ^_^

**Comment by eLBati** [comment]
> Se in questa situazione entra in gioco anche un protagonista abbastanza sordo come Odoo SA allora la cosa si complica un po'.

@OpenCode le esperienze passate non sono state molto costruttive, però in questo momento c'è molto interesse da parte loro nel progetto "localizzazione italiana"; quindi dal mio punto di vista ci sono buone probabilità che ascoltino di più, o almeno può valer la pena provarci.
Inoltre i contributori italiani sono più numerosi e più competenti rispetto a qualche anno fa; anche questo dovrebbe semplificare il lavoro.

> La strada che stanno intraprendendo, noi l'abbiamo già usata con tutti i suoi limiti nel passato

Però mi pare che la loro proposta corrisponda a uno dei casi d'uso che intendiamo coprire con il refactoring, vedi "DDT semplificato" del doc.

> Servirebbe che si fermassero e ci coinvolgessero nella fase di analisi preliminare. Possiamo portare migliaia di ore complessive di esperienza sia tecnica che funzionale su un argomento di cui loro conoscono a mala pena il significato

Questa è  la proposta da fargli.
Bisognerebbe iniziare testando e revisionando quello che hanno fatto, individuando i casi d'uso coperti, quelli non coperti e quelli eventualmente coperti male.
Questi utlimi andranno corretti da loro.
Per quelli non coperti si valuterà anche con loro se farglieli coprire o metterli in moduli OCA.

**Comment by eLBati** [comment]
https://github.com/odoo/odoo/pull/32156#issuecomment-581408705

La mancanza di risposta la prendiamo come un "non lo faremo"?

**Comment by robyf70** [comment]
... dopo la foga iniziale tutto si calma?
